### PR TITLE
added computingEnvironmentsUsed datatype property

### PR DESCRIPTION
**Pull request by zednis**
references USGCRP/gcis/issues/150

### CODE REVIEW

**Comment by justgo129** [comment]
Looks good to me.

On Tue, Jun 23, 2015 at 1:31 PM, Stephan Zednik notifications@github.com
wrote:

> ## references USGCRP/gcis#150 https://github.com/USGCRP/gcis/issues/150
> 
> You can view, comment on, or merge this pull request online at:
> 
>   https://github.com/USGCRP/gcis-ontology/pull/21
> Commit Summary
> - added computingEnvironmentsUsed datatype property
> 
> File Changes
> - _M_ gcis.ttl
>   https://github.com/USGCRP/gcis-ontology/pull/21/files#diff-0 (6)
> 
> Patch Links:
> - https://github.com/USGCRP/gcis-ontology/pull/21.patch
> - https://github.com/USGCRP/gcis-ontology/pull/21.diff
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/pull/21.

## 

---

Justin Goldstein, Ph.D.
Advance Science Climate Data and Observing Systems Coordinator
US Global Change Research Program
1717 Pennsylvania Ave NW, Suite #250
Washington, DC 20006

O: (202) 419-3496
M: (202) 285-3005

e-mail: jgoldstein AT usgcrp Dot gov
http://www.globalchange.gov

**Comment by justgo129** [comment]
Actually, I take that back.  @zednis , could you please clarify the value of the comment?  I don't understand the latter portion.  Do you mean the software agents associated "with" an activity, or the associations between one software agent and another?  I think we want the first but it reads like the second as currently written.  Maybe it's just me.

**Comment by zednis** [comment]
@justgo129 

> Do you mean the software agents associated "with" an activity, or the associations between one software agent and another?

The first.  The computing environments are related to the software agents that are associated with the activity.

example - if IDL 8.0 is a software agent associated with the activity then the computing environment - OSX 10.9 - describes the platform on which IDL 8.0 was being executed.

If we had a 1-to-1 mapping of a specific agent to its computing platform we could model it like this:

```
prov:qualifiedAssociation [
      a prov:Association ;
      prov:agent [ 
         a prov:SoftwareAgent, gcis:Software, dbpedia-owl:Software ;
         rdfs:label "IDL (version 8.0)"^xsd:string ;
         dbpedia-owl:computingPlatform [ rdfs:label "Mac OS X (darwin x86_64 m64)"^^xsd:string; ] ;
      ] ;
      prov:hadPlan [ .. ] ;
   ] ;

```

Unfortunately we don't have the 1-to-1 mapping so we can't confidently represent it like this without including computing platform info that is not relevant for the agent.

Instead I have proposed the creation a new property that has represents a description of one or more computing platforms for one or more agents associated with the activity.

example:

```
<http://data.globalchange.gov/activity/0158fa86-nca3-ghcn-daily-r201305-process>
  a prov:Activity ;
  gcis:computingEnvironmentsUsed "Linux (CentOS release 6.4); Mac OS X (darwin x86_64 m64)"^^xsd:string .
```

**Comment by justgo129** [comment]
Makes perfect sense to me.  If possible, I'd still like to see some changes
in the rdf:comment for the item in the pull request though.  Thanks.

On Tue, Jun 23, 2015 at 2:49 PM, Stephan Zednik notifications@github.com
wrote:

> @justgo129 https://github.com/justgo129
> 
> Do you mean the software agents associated "with" an activity, or the
> associations between one software agent and another?
> 
> The first. The computing environments are related to the software agents
> that are associated with the activity.
> 
> example - if IDL 8.0 is a software agent associated with the activity then
> the computing environment - OSX 10.9 - describes the platform on which IDL
> 8.0 was being executed.
> 
> If we had a 1-to-1 mapping of a specific agent to its computing platform
> we could model it like this:
> 
> prov:qualifiedAssociation [
>       a prov:Association ;
>       prov:agent [
>          a prov:SoftwareAgent, gcis:Software, dbpedia-owl:Software ;
>          rdfs:label "IDL (version 8.0)"^xsd:string ;
>          dbpedia-owl:computingPlatform [ rdfs:label "Mac OS X (darwin x86_64 m64)"^^xsd:string; ] ;
>       ] ;
>       prov:hadPlan [ .. ] ;
>    ] ;
> 
> Unfortunately we don't have the 1-to-1 mapping so we can't model it like
> this without including computing platform info that is possibly not
> relevant for the agent.
> 
> Instead I have proposed the creation a new property that has represents a
> description of one or more computing platforms for one or more agents
> associated with the activity.
> 
> example:
> 
> http://data.globalchange.gov/activity/0158fa86-nca3-ghcn-daily-r201305-process
>   a prov:Activity ;
>   gcis:computingEnvironmentsUsed "Linux (CentOS release 6.4); Mac OS X (darwin x86_64 m64)"^^xsd:string .
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114605870.

## 

---

Justin Goldstein, Ph.D.
Advance Science Climate Data and Observing Systems Coordinator
US Global Change Research Program
1717 Pennsylvania Ave NW, Suite #250
Washington, DC 20006

O: (202) 419-3496
M: (202) 285-3005

e-mail: jgoldstein AT usgcrp Dot gov
http://www.globalchange.gov

**Comment by zednis** [comment]
Oh, ok.  Do you have any suggestions for how we can make the property's comment more clear?

**Comment by justgo129** [comment]
What do you think of the following?

rdfs:comment "Titles and versions of the computing environments associated
with the software agents involved in an activity";

We don't always have versions though.

Currently, the comment reads:

rdfs:comment "Description of the computing environments of software
agents associated in an activity" ;

On Tue, Jun 23, 2015 at 3:45 PM, Stephan Zednik notifications@github.com
wrote:

> Oh, ok. Do you have any suggestions for how we can make the property's
> comment more clear?
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114621393.

## 

---

Justin Goldstein, Ph.D.
Advance Science Climate Data and Observing Systems Coordinator
US Global Change Research Program
1717 Pennsylvania Ave NW, Suite #250
Washington, DC 20006

O: (202) 419-3496
M: (202) 285-3005

e-mail: jgoldstein AT usgcrp Dot gov
http://www.globalchange.gov

**Comment by rewolfe** [comment]
What about: "Description of the software and computing environments
associated with an activity."

On Tue, Jun 23, 2015 at 4:07 PM, justgo129 notifications@github.com wrote:

> What do you think of the following?
> 
> rdfs:comment "Titles and versions of the computing environments associated
> with the software agents involved in an activity";
> 
> We don't always have versions though.
> 
> Currently, the comment reads:
> 
> rdfs:comment "Description of the computing environments of software
> agents associated in an activity" ;
> 
> On Tue, Jun 23, 2015 at 3:45 PM, Stephan Zednik notifications@github.com
> wrote:
> 
> > Oh, ok. Do you have any suggestions for how we can make the property's
> > comment more clear?
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114621393
> > .
> 
> ## 
> 
> ---
> 
> Justin Goldstein, Ph.D.
> Advance Science Climate Data and Observing Systems Coordinator
> US Global Change Research Program
> 1717 Pennsylvania Ave NW, Suite #250
> Washington, DC 20006
> 
> O: (202) 419-3496
> M: (202) 285-3005
> 
> e-mail: jgoldstein AT usgcrp Dot gov
> http://www.globalchange.gov
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114625767.

## 

Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966

**Comment by justgo129** [comment]
Works for me.

On Wed, Jun 24, 2015 at 9:47 AM, Robert Wolfe notifications@github.com
wrote:

> What about: "Description of the software and computing environments
> associated with an activity."
> 
> On Tue, Jun 23, 2015 at 4:07 PM, justgo129 notifications@github.com
> wrote:
> 
> > What do you think of the following?
> > 
> > rdfs:comment "Titles and versions of the computing environments
> > associated
> > with the software agents involved in an activity";
> > 
> > We don't always have versions though.
> > 
> > Currently, the comment reads:
> > 
> > rdfs:comment "Description of the computing environments of software
> > agents associated in an activity" ;
> > 
> > On Tue, Jun 23, 2015 at 3:45 PM, Stephan Zednik <
> > notifications@github.com>
> > wrote:
> > 
> > > Oh, ok. Do you have any suggestions for how we can make the property's
> > > comment more clear?
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> > > https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114621393
> > > .
> > 
> > ## 
> > 
> > ---
> > 
> > Justin Goldstein, Ph.D.
> > Advance Science Climate Data and Observing Systems Coordinator
> > US Global Change Research Program
> > 1717 Pennsylvania Ave NW, Suite #250
> > Washington, DC 20006
> > 
> > O: (202) 419-3496
> > M: (202) 285-3005
> > 
> > e-mail: jgoldstein AT usgcrp Dot gov
> > http://www.globalchange.gov
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114625767
> > .
> 
> ## 
> 
> Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114872915.

## 

---

Justin Goldstein, Ph.D.
Advance Science Climate Data and Observing Systems Coordinator
US Global Change Research Program
1717 Pennsylvania Ave NW, Suite #250
Washington, DC 20006

O: (202) 419-3496
M: (202) 285-3005

e-mail: jgoldstein AT usgcrp Dot gov
http://www.globalchange.gov

**Comment by justgo129** [comment]
At a second look, I am fine with it but am wondering whether we could avoid
using the term in its definition.

On Wed, Jun 24, 2015 at 10:12 AM, Justin Goldstein jgoldstein@usgcrp.gov
wrote:

> Works for me.
> 
> On Wed, Jun 24, 2015 at 9:47 AM, Robert Wolfe notifications@github.com
> wrote:
> 
> > What about: "Description of the software and computing environments
> > associated with an activity."
> > 
> > On Tue, Jun 23, 2015 at 4:07 PM, justgo129 notifications@github.com
> > wrote:
> > 
> > > What do you think of the following?
> > > 
> > > rdfs:comment "Titles and versions of the computing environments
> > > associated
> > > with the software agents involved in an activity";
> > > 
> > > We don't always have versions though.
> > > 
> > > Currently, the comment reads:
> > > 
> > > rdfs:comment "Description of the computing environments of software
> > > agents associated in an activity" ;
> > > 
> > > On Tue, Jun 23, 2015 at 3:45 PM, Stephan Zednik <
> > > notifications@github.com>
> > > wrote:
> > > 
> > > > Oh, ok. Do you have any suggestions for how we can make the property's
> > > > comment more clear?
> > > > 
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > > > https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114621393
> > > > .
> > > 
> > > ## 
> > > 
> > > ---
> > > 
> > > Justin Goldstein, Ph.D.
> > > Advance Science Climate Data and Observing Systems Coordinator
> > > US Global Change Research Program
> > > 1717 Pennsylvania Ave NW, Suite #250
> > > Washington, DC 20006
> > > 
> > > O: (202) 419-3496
> > > M: (202) 285-3005
> > > 
> > > e-mail: jgoldstein AT usgcrp Dot gov
> > > http://www.globalchange.gov
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114625767
> > > .
> > 
> > ## 
> > 
> > Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114872915.
> 
> ## 
> 
> ---
> 
> Justin Goldstein, Ph.D.
> Advance Science Climate Data and Observing Systems Coordinator
> US Global Change Research Program
> 1717 Pennsylvania Ave NW, Suite #250
> Washington, DC 20006
> 
> O: (202) 419-3496
> M: (202) 285-3005
> 
> e-mail: jgoldstein AT usgcrp Dot gov
> http://www.globalchange.gov

## 

---

Justin Goldstein, Ph.D.
Advance Science Climate Data and Observing Systems Coordinator
US Global Change Research Program
1717 Pennsylvania Ave NW, Suite #250
Washington, DC 20006

O: (202) 419-3496
M: (202) 285-3005

e-mail: jgoldstein AT usgcrp Dot gov
http://www.globalchange.gov

**Comment by rewolfe** [comment]
We are not defining a term, but a resource.  Suppose we had property of a
Person called firstName, wouldn't it be define as "The person's first name".

On Wed, Jun 24, 2015 at 10:13 AM, justgo129 notifications@github.com
wrote:

> At a second look, I am fine with it but am wondering whether we could avoid
> using the term in its definition.
> 
> On Wed, Jun 24, 2015 at 10:12 AM, Justin Goldstein jgoldstein@usgcrp.gov
> wrote:
> 
> > Works for me.
> > 
> > On Wed, Jun 24, 2015 at 9:47 AM, Robert Wolfe notifications@github.com
> > wrote:
> > 
> > > What about: "Description of the software and computing environments
> > > associated with an activity."
> > > 
> > > On Tue, Jun 23, 2015 at 4:07 PM, justgo129 notifications@github.com
> > > wrote:
> > > 
> > > > What do you think of the following?
> > > > 
> > > > rdfs:comment "Titles and versions of the computing environments
> > > > associated
> > > > with the software agents involved in an activity";
> > > > 
> > > > We don't always have versions though.
> > > > 
> > > > Currently, the comment reads:
> > > > 
> > > > rdfs:comment "Description of the computing environments of software
> > > > agents associated in an activity" ;
> > > > 
> > > > On Tue, Jun 23, 2015 at 3:45 PM, Stephan Zednik <
> > > > notifications@github.com>
> > > > wrote:
> > > > 
> > > > > Oh, ok. Do you have any suggestions for how we can make the
> > > > > property's
> > > > > comment more clear?
> > > > > 
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <
> > > > > https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114621393
> > > > > .
> > > > 
> > > > ## 
> > > > 
> > > > ---
> > > > 
> > > > Justin Goldstein, Ph.D.
> > > > Advance Science Climate Data and Observing Systems Coordinator
> > > > US Global Change Research Program
> > > > 1717 Pennsylvania Ave NW, Suite #250
> > > > Washington, DC 20006
> > > > 
> > > > O: (202) 419-3496
> > > > M: (202) 285-3005
> > > > 
> > > > e-mail: jgoldstein AT usgcrp Dot gov
> > > > http://www.globalchange.gov
> > > > 
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > > > https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114625767
> > > > .
> > > 
> > > ## 
> > > 
> > > Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966
> > > 
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114872915
> > > .
> > 
> > ## 
> > 
> > ---
> > 
> > Justin Goldstein, Ph.D.
> > Advance Science Climate Data and Observing Systems Coordinator
> > US Global Change Research Program
> > 1717 Pennsylvania Ave NW, Suite #250
> > Washington, DC 20006
> > 
> > O: (202) 419-3496
> > M: (202) 285-3005
> > 
> > e-mail: jgoldstein AT usgcrp Dot gov
> > http://www.globalchange.gov
> 
> ## 
> 
> ---
> 
> Justin Goldstein, Ph.D.
> Advance Science Climate Data and Observing Systems Coordinator
> US Global Change Research Program
> 1717 Pennsylvania Ave NW, Suite #250
> Washington, DC 20006
> 
> O: (202) 419-3496
> M: (202) 285-3005
> 
> e-mail: jgoldstein AT usgcrp Dot gov
> http://www.globalchange.gov
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114881427.

## 

Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966

**Comment by rewolfe** [comment]
I meant "We are not defining a term, but a property."

On Wed, Jun 24, 2015 at 1:08 PM, Robert Wolfe rewolfe@usgcrp.gov wrote:

> We are not defining a term, but a resource.  Suppose we had property of a
> Person called firstName, wouldn't it be define as "The person's first name".
> 
> On Wed, Jun 24, 2015 at 10:13 AM, justgo129 notifications@github.com
> wrote:
> 
> > At a second look, I am fine with it but am wondering whether we could
> > avoid
> > using the term in its definition.
> > 
> > On Wed, Jun 24, 2015 at 10:12 AM, Justin Goldstein <jgoldstein@usgcrp.gov
> > 
> > > wrote:
> > > 
> > > Works for me.
> > > 
> > > On Wed, Jun 24, 2015 at 9:47 AM, Robert Wolfe <notifications@github.com
> > > 
> > > wrote:
> > > 
> > > > What about: "Description of the software and computing environments
> > > > associated with an activity."
> > > > 
> > > > On Tue, Jun 23, 2015 at 4:07 PM, justgo129 notifications@github.com
> > > > wrote:
> > > > 
> > > > > What do you think of the following?
> > > > > 
> > > > > rdfs:comment "Titles and versions of the computing environments
> > > > > associated
> > > > > with the software agents involved in an activity";
> > > > > 
> > > > > We don't always have versions though.
> > > > > 
> > > > > Currently, the comment reads:
> > > > > 
> > > > > rdfs:comment "Description of the computing environments of software
> > > > > agents associated in an activity" ;
> > > > > 
> > > > > On Tue, Jun 23, 2015 at 3:45 PM, Stephan Zednik <
> > > > > notifications@github.com>
> > > > > wrote:
> > > > > 
> > > > > > Oh, ok. Do you have any suggestions for how we can make the
> > > > > > property's
> > > > > > comment more clear?
> > > > > > 
> > > > > > —
> > > > > > Reply to this email directly or view it on GitHub
> > > > > > <
> > > > > > https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114621393
> > > > > > .
> > > > > 
> > > > > ## 
> > > > > 
> > > > > ---
> > > > > 
> > > > > Justin Goldstein, Ph.D.
> > > > > Advance Science Climate Data and Observing Systems Coordinator
> > > > > US Global Change Research Program
> > > > > 1717 Pennsylvania Ave NW, Suite #250
> > > > > Washington, DC 20006
> > > > > 
> > > > > O: (202) 419-3496
> > > > > M: (202) 285-3005
> > > > > 
> > > > > e-mail: jgoldstein AT usgcrp Dot gov
> > > > > http://www.globalchange.gov
> > > > > 
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <
> > > > > https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114625767
> > > > > .
> > > > 
> > > > ## 
> > > > 
> > > > Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966
> > > > 
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > > > https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114872915>.
> > > 
> > > ## 
> > > 
> > > ---
> > > 
> > > Justin Goldstein, Ph.D.
> > > Advance Science Climate Data and Observing Systems Coordinator
> > > US Global Change Research Program
> > > 1717 Pennsylvania Ave NW, Suite #250
> > > Washington, DC 20006
> > > 
> > > O: (202) 419-3496
> > > M: (202) 285-3005
> > > 
> > > e-mail: jgoldstein AT usgcrp Dot gov
> > > http://www.globalchange.gov
> > 
> > ## 
> > 
> > ---
> > 
> > Justin Goldstein, Ph.D.
> > Advance Science Climate Data and Observing Systems Coordinator
> > US Global Change Research Program
> > 1717 Pennsylvania Ave NW, Suite #250
> > Washington, DC 20006
> > 
> > O: (202) 419-3496
> > M: (202) 285-3005
> > 
> > e-mail: jgoldstein AT usgcrp Dot gov
> > http://www.globalchange.gov
> > 
> > —
> > Reply to this email directly or view it on GitHub
> > https://github.com/USGCRP/gcis-ontology/pull/21#issuecomment-114881427.
> 
> ## 
> 
> Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966

## 

Robert Wolfe, NASA GSFC @ USGCRP, o: 202-419-3470, m: 301-257-6966

**Comment by justgo129** [comment]
I see your point, Robert, and concur unless @zednis feels differently.  It's one thing if the item is a class, and another thing if it's a property.

**Comment by zednis** [comment]
I am hesitant to include "software and computing environment" since we have another field/property for software and this property is intended specifically for the computing environment.

**Comment by justgo129** [comment]
So, to adapt Robert's suggestion:
""Description of the computing environment(s)
associated with an activity."?

I like that.  @zednis @rewolfe what do you think?

**Comment by zednis** [comment]
I am ok with that

**Comment by rewolfe** [comment]
Cool. +1

**Comment by zednis** [comment]
comment updated

**Comment by zednis** [comment]
Are we ready to merge this update?

**Comment by justgo129** [comment]
Fine with me.

File: gcis.ttl [hunk 1]
```diff
@@ -1033,3 +1033,9 @@ gcis:isSupportedBy  a owl:ObjectProperty ;
 	rdfs:label "Is Supported By" ;
 	rdfs:comment "A finding is supported by data, models, model runs, software, etc.";
 	rdfs:domain gcis:Finding .	
+
+gcis:computingEnvironmentsUsed a owl:DatatypeProperty ;
+	rdfs:label "computing environments used" ;
```
Comment (bduggan) [inline gcis.ttl:1036]: This should be "Computing Environments Used" so that the capitalization is consistent with the other rdfs:label's.

Comment (zednis) [inline gcis.ttl:1036]: It is conventional in RDF for rdfs:labels on properties to be lower-case.  I believe this is a case where the convention GCIS was using is not standard.

Comment (bduggan) [inline gcis.ttl:1036]: On Monday, July  6, Stephan Zednik wrote: 

> > @@ -1033,3 +1033,9 @@ gcis:isSupportedBy  a owl:ObjectProperty ;
> >     rdfs:label "Is Supported By" ;
> >     rdfs:comment "A finding is supported by data, models, model runs, software, etc.";
> >     rdfs:domain gcis:Finding .  
> > +
> > +gcis:computingEnvironmentsUsed a owl:DatatypeProperty ;
> > -   rdfs:label "computing environments used" ;
> 
> It is conventional in RDF for rdfs:labels on properties to be lower-case.  I believe this is a case where the convention GCIS was using is not standard.

Sounds good to me, in that case let's merge it and I will make another
ticket to make all of the other rdfs:labels lower-case.

Comment (zednis) [inline gcis.ttl:1036]: :+1:

**Pull request merged**